### ISSUE TITLE
LR Upload invalid source rocks.

### ISSUE THREAD

**Issue by moteus** [keplerproject/luarocks#299]
I jast start using `luarocks upload` and facing with porblem that on server uploading invalid rocks files.
For example you can see [try](https://rocks.moonscript.org/manifests/moteus/try-0.1.0-1.src.rock) module.
Command I use is just `luarocks upload rockspecs\try-0.1.0-1.rockspec`
```
Packing try
Sending F:\e\Projects\lua-try/try-0.1.0-1.src.rock ...
Done: http://rocks.moonscript.org/modules/moteus/try
```
Result `.src.rock` file is correct and I can just upload it manually via web UI.

I use luarocks master on Windows XP.

**Comment by hishamhm** [comment]
`luarocks upload` was never really tested on Windows. I haven't been able to take a closer look at this, help is welcome! @ignacio, @Tieske?

**Comment by ignacio** [comment]
There are some things left to work out regarding uploads (see #273).
In this case, I tried to reproduce by downloading the source and uploading. I used curl and it worked fine, but as per #273, it does not work reliably.

**Comment by moteus** [comment]
Problem in https://github.com/keplerproject/luarocks/blob/master/src/luarocks/upload/multipart.lua#L29
When you try upload zip file. Please add `rb` open file mode.

**Comment by ignacio** [comment]
Pushed a change to master. Can you confirm it fixes your issue?

**Comment by moteus** [comment]
This changes fix problen on my installation of LR 2.2.0. So I think it should also works on 2.2.1.
